Download.cs

```diff
@@ -85,6 +85,11 @@ public class Download
             {
                 Console.WriteLine(ex.Message);
             }
+            catch (Exception ex)
+            {
+                ex = (ex.InnerException != null) ? ex.InnerException.GetBaseException() : ex;
+                Console.WriteLine(ex.Message);

```

Comment:
Minor - unless you're developing a UI app, it's almost always better for us to use ex.ToString(), rather than ex.Message().  ToString() contains the stack trace, and calls ToString() on the InnerException